compiler/transforms/schedule_memref_linalg.py

```diff
@@ -0,0 +1,302 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, memref_stream, stream
+from xdsl.ir import Block, Region
+from xdsl.ir.affine import AffineConstantExpr, AffineDimExpr, AffineExpr, AffineMap
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint
+
+from compiler.util.canonicalize_affine import canonicalize_map
+
+
+def disable_dims(map: AffineMap, dim: int) -> AffineMap:
+    """
+    Returns an affine map with the first `dim` dimensions set to 0
+
+    For example:
+        (d0, d1, d2) -> d0 + d1 + d2
+    For `dim` = 1, will return:
+        (d1, d2) -> d1 + d2
+    For `dim` = 2, will return:
+        (d2) -> d2
+    """
+    return canonicalize_map(
+        map.replace_dims_and_symbols(
+            [AffineConstantExpr(0) for _ in range(dim)]
+            + [AffineDimExpr(i) for i in range(map.num_dims - dim)],
+            [],
+            map.num_dims - dim,
+            0,
```

Comment:
Does this mean we should we also add todo's here for adding dynamic shapes?

Reply:
yes 😢